weldx/core.py

```diff
@@ -0,0 +1,492 @@
+                return False
+            return (
+                np.all(self.data == other.data)
+                and np.all(self.time == other.time)
+                and self._interpolation == other.interpolation
+            )
+
+        return self._data == other.data
+
+    def __repr__(self):
+        """Give __repr__ output."""
+        representation = "<TimeSeries>"
+        if isinstance(self._data, xr.DataArray):
+            if self._data.shape[0] == 1:
+                representation += f"\nConstant value:\n\t{self.data.magnitude[0]}\n"
+            else:
+                representation += (
+                    f"\nTime:\n\t{self.time}\n"
+                    + f"Values:\n\t{self.data.magnitude}\n"
+                    + f"Interpolation:\n\t{self.interpolation}\n"
+                )
+        else:
+            representation += self.data.__repr__().replace(
+                "<MathematicalExpression>", ""
+            )
+        return representation + f"Units:\n\t{self.units}\n"
+
+    @property
+    def data(self) -> Union[xr.DataArray, MathematicalExpression]:
+        """Return the data of the TimeSeries.
+
+        This is either a set of discrete values or a mathematical expression.
+
+        Returns
+        -------
+        xr.DataArray:
+            Underlying data array with discrete values in time.
+        MathematicalExpression:
+            A mathematical expression describing the time dependency
+
+        """
+        if isinstance(self._data, xr.DataArray):
+            return self._data.data
+        return self._data
+
+    @property
+    def interpolation(self) -> Union[str, None]:
+        """Return the interpolation.
+
+        Returns
+        -------
+        str:
+            Interpolation of the TimeSeries
+
+        """
+        return self._interpolation
+
+    @property
+    def time(self) -> Union[None, pd.TimedeltaIndex]:
+        """Return the data's timestamps.
+
+        Returns
+        -------
+        pd.TimedeltaIndex:
+            Timestamps of the  data
+
+        """
+        if isinstance(self._data, xr.DataArray):
+            if len(self._data.time) == 1:
+                return None
+            return ut.to_pandas_time_index(self._data.time.data)
+        return self._time
+
+    def interp_time(self, time: Union[pd.TimedeltaIndex, pint.Quantity]):
+        """Interpolate the TimeSeries in time.
+
+        If the internal data consists of discrete values, an interpolation with the
+        prescribed interpolation method is performed. In case of mathematical
+        expression, the expression is evaluated for the given timestamps.
+
+        Parameters
+        ----------
+        time:
+            A set of timestamps.
+
+        Returns
+        -------
+        xr.DataArray:
+            A data array containing the interpolated data.
+
+        """
+        if isinstance(self._data, xr.DataArray):
+            # constant values are also treated by this branch
```

Comment:
I think we need a check (and conversion) if `time` is `pint.Quantity` here?

Reply:
Done